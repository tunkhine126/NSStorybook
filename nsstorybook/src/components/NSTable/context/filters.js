import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

export const NSFiltersContext = createContext();

export const NSFiltersProvider = ({
  children,
  filterCategories,
  lastUpdateProps,
  dateRangeProps,
  handleSelected,
}) => {
  const types = {
    multi: {},
    levels: {},
    single: {},
    date: {},
  };

  const { lastUpdate } = lastUpdateProps;

  ['multi', 'levels', 'single', 'date'].forEach(type => {
    filterCategories
      .filter(c => c.type === type)
      .forEach(({ id }) => {
        types[type][id] = [];
      });
  });

  const [singleSelectOptions, setSingleSelectOptions] = useState(types.single);
  const [multiSelectOptions, setMultiSelectOptions] = useState(types.multi);
  const [levelSelectOptions, setLevelSelectOptions] = useState(types.levels);
  const [showCalendar, setShowCalendar] = useState(
    lastUpdate.query === 'custom'
  );
  const [localDateRange, setlocalDateRange] = useState({
    before: null,
    after: null,
  });

  return (
    <NSFiltersContext.Provider
      value={{
        multiSelect: { multiSelectOptions, setMultiSelectOptions },
        levelSelect: { levelSelectOptions, setLevelSelectOptions },
        singleSelect: { singleSelectOptions, setSingleSelectOptions },
        calendarProps: { showCalendar, setShowCalendar },
        localDateProps: { localDateRange, setlocalDateRange },
        lastUpdateProps,
        dateRangeProps,
        filterCategories,
        handleSelected,
      }}
    >
      {children}
    </NSFiltersContext.Provider>
  );
};

NSFiltersProvider.propTypes = {
  children: PropTypes.any.isRequired,
  dateRangeProps: PropTypes.object.isRequired,
  lastUpdateProps: PropTypes.object.isRequired,
  filterCategories: PropTypes.array.isRequired,
  handleSelected: PropTypes.func.isRequired,
};
