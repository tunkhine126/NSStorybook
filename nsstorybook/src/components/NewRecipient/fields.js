/**
 *
 * Fields
 *
 */

import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Grid from '@material-ui/core/Grid';

import NSFields from '../NSFields';

import { required } from 'utils/validations';
import { commonStyles } from './styles';

export function Fields({
  fields,
  showPreFixId,
  preFixSeparator,
  parentInternalId,
  preSetValue,
  fieldsRequired = false,
}) {
  let requiredProps = {};
  const classes = commonStyles();

  if (fieldsRequired) {
    requiredProps = {
      validateOnChange: true,
      validateOnBlur: true,
      required: true,
      validate: required,
    };
  }

  const [fieldState, setFieldState] = useState(
    fields.reduce((o, key) => ({ ...o, [key.name]: { item: [] } }), {})
  );

  const handleFieldChange = (field, value) =>
    setFieldState(state => ({ ...state, [field]: { item: value } }));

  const Prefix = () => (
    <p style={{ width: 'max-content' }}>{parentInternalId + preFixSeparator}</p>
  );

  const fieldsToRender =
    fields.length > 0 &&
    fields.map((f, idx) => {
      const selectFields = ['single_select', 'multi_select', 'likert'];
      if (selectFields.includes(f.field_type)) {
        return (
          <NSFields
            key={idx}
            fieldType={f.field_type}
            field={`${f.name}_custom`}
            label={f.contexts.us_en.label}
            values={fieldState[f.name]}
            options={f.option_values.map(o => ({ value: o }))}
            onchange={v => handleFieldChange(f.name, v)}
            {...requiredProps}
          />
        );
      }
      return (
        <NSFields
          key={idx}
          fieldType={f.field_type}
          field={`${f.name}_custom`}
          label={f.contexts.us_en.label}
          {...requiredProps}
        />
      );
    });
  return (
    <Grid item className={classes.spacing}>
      {fieldsRequired && (
        <NSFields
          fieldType="short_text"
          field="name"
          label="Name"
          {...requiredProps}
        />
      )}

      {!fieldsRequired && (
        <>
          <NSFields
            fieldType="photo"
            field="base64Avatar"
            label="Avatar"
            {...requiredProps}
          />

          <NSFields
            fieldType="short_text"
            field="internalId"
            label="Internal Id"
            position="start"
            initialValue={preSetValue}
            showPrefix={showPreFixId}
            icon={Prefix}
            {...requiredProps}
          />

          <NSFields
            fieldType="paragraph_text"
            field="description"
            label="Description"
            {...requiredProps}
          />
        </>
      )}

      {fieldsToRender}
    </Grid>
  );
}

Fields.propTypes = {
  fields: PropTypes.array.isRequired,
  fieldsRequired: PropTypes.bool,
  showPreFixId: PropTypes.bool,
  preFixSeparator: PropTypes.string,
  parentInternalId: PropTypes.string,
  preSetValue: PropTypes.string,
};

export default Fields;
