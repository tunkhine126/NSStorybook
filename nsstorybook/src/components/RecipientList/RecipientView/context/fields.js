import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const FieldsContext = createContext();

export const FieldsProvider = ({ children, ...rest }) => {
  const [activePhotoField, setActivePhotoField] = useState(null);
  const [recipientPhoto, setRecipientPhoto] = useState({});

  return (
    <FieldsContext.Provider
      value={{
        activePhotoProps: { activePhotoField, setActivePhotoField },
        recipientPhotoProps: { recipientPhoto, setRecipientPhoto },
        rest,
      }}
    >
      {children}
    </FieldsContext.Provider>
  );
};

FieldsProvider.propTypes = {
  children: PropTypes.any.isRequired,
};
