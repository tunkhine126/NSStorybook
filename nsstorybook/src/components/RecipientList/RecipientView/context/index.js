import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const RecipientViewContext = createContext();

export const RecipientViewProvider = ({
  children,
  tabChange,
  rdData,
  data,
  profileDetails,
  primaryRecipient,
  match,
  ...rest
}) => {
  const [toaster, setToaster] = useState({
    show: false,
    recipientUuid: null,
    field: null,
    undoHandler: null,
  });
  const [saveState, setSaveState] = useState({});

  return (
    <RecipientViewContext.Provider
      value={{
        data,
        rdData,
        tabChange,
        profileDetails,
        primaryRecipient,
        match,
        toasterProps: { toaster, setToaster },
        saveProps: { saveState, setSaveState },
        rest,
      }}
    >
      {children}
    </RecipientViewContext.Provider>
  );
};

RecipientViewProvider.propTypes = {
  children: PropTypes.any.isRequired,
  tabChange: PropTypes.func.isRequired,
  profileDetails: PropTypes.object.isRequired,
  primaryRecipient: PropTypes.bool.isRequired,
  data: PropTypes.object.isRequired,
  rdData: PropTypes.any.isRequired,
  match: PropTypes.any.isRequired,
};
