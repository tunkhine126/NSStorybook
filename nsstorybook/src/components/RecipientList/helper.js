import React from 'react';
import clsx from 'clsx';
import moment from 'moment';

import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';

import LinkRouter from 'components/shared/LinkRouter';

export const formattedColumns = (classes, columns) =>
  columns
    .filter(c => !c.noColumn)
    .map((c, idx) => ({
      direction: idx === 0 ? 'column' : 'row',
      justify: idx < 2 ? 'flex-start' : 'center',
      alignItems: 'flex-start',
      columnId: c.id,
      loadingRows: [{}, {}],
      content(row) {
        const rowData = row[c.id];

        return idx === 0 ? (
          <>
            <Grid item>
              <LinkRouter
                color="primary"
                className={clsx(classes.recipientLink, 'recipientColumn')}
                to={row.recipientPath.replace(':id', row.uuid)}
              >
                {rowData}
              </LinkRouter>
            </Grid>
            {row.parentRecipients && (
              <Grid item className={classes.familyCommunity}>
                <Typography variant="caption" className={classes.caption}>
                  {row.parentRecipients}
                </Typography>
              </Grid>
            )}
          </>
        ) : (
          <Grid item className={classes.columnHeight}>
            {c.type === 'date'
              ? moment(rowData).format('MMMM DD, YYYY')
              : rowData}
          </Grid>
        );
      },
    }));

export const generateColumns = (data, columns) =>
  columns.map(column => {
    const columnData = data.find(d => d.id === column.id);
    if (columnData) {
      column.columnData = columnData;
    }
    return column;
  });
