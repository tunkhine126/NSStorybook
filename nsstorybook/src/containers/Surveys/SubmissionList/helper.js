import React from 'react';
import clsx from 'clsx';
import moment from 'moment';

import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Avatar from '@material-ui/core/Avatar';
import Chip from '@material-ui/core/Chip';

import LinkRouter from 'components/shared/LinkRouter';
import { ROUTES } from '../../../utils/globalConstants';

export function formatData(
  family,
  community,
  id,
  surveyId,
  recipientUuid,
  familyId,
  survey,
  surveyor,
  completed
) {
  return {
    family,
    community,
    id,
    surveyId,
    recipientUuid,
    familyId,
    survey,
    surveyor,
    completed,
  };
}

export const formattedColumns = (classes, primaryRecipientIdPath) => [
  {
    direction: 'column',
    justify: 'flex-start',
    alignItems: 'flex-start',
    columnId: 'family',
    loadingRows: [{}, {}],
    content(row) {
      return (
        <>
          <Grid item>
            <LinkRouter
              color="primary"
              className={clsx(classes.recipientLink)}
              to={primaryRecipientIdPath.replace(
                ':id',
                `${row.recipientUuid}?tab=2&name=${row.survey}&date=${
                  row.completed
                }`
              )}
            >
              {row.family}
            </LinkRouter>
          </Grid>

          <Grid item className={classes.familyCommunity}>
            <Typography variant="caption" className={classes.caption}>
              {row.community}
            </Typography>
          </Grid>
        </>
      );
    },
  },
  {
    direction: 'row',
    justify: 'flex-start',
    alignItems: 'flex-start',
    columnId: 'id',
    loadingRows: [{}],
    content(row) {
      return (
        <Grid item className={classes.columnHeight}>
          {row.familyId}
        </Grid>
      );
    },
  },
  {
    direction: 'row',
    justify: 'flex-start',
    alignItems: 'flex-start',
    columnId: 'survey',
    loadingRows: [{}],
    content(row) {
      return (
        <Grid item className={classes.columnHeight}>
          <LinkRouter
            color="primary"
            to={ROUTES.surveys.children[1].path.replace(':id', row.surveyId)}
          >
            {row.survey}
          </LinkRouter>
        </Grid>
      );
    },
  },
  {
    direction: 'row',
    justify: 'flex-start',
    alignItems: 'flex-start',
    columnId: 'surveyor',
    loadingRows: [{}],
    content(row) {
      return (
        <Grid item className={classes.columnHeight}>
          <Chip
            component="span"
            avatar={
              <Avatar className={classes.avatar} component="span">
                {row.surveyor.split('')[0]}
              </Avatar>
            }
            label={row.surveyor}
            className={classes.chip}
          />
        </Grid>
      );
    },
  },
  {
    direction: 'row',
    justify: 'flex-start',
    alignItems: 'flex-start',
    columnId: 'completed',
    loadingRows: [{}],
    content(row) {
      return (
        <>
          <Grid item xs={12}>
            {moment(row.completed).format('MMMM DD, YYYY')}
          </Grid>
          <Grid item className={classes.caption}>
            {moment(row.completed)
              .startOf('hour')
              .fromNow()}
          </Grid>
        </>
      );
    },
  },
];

export const columns = (
  label,
  { surveyData, surveyorData, communitiesData, completedData }
) => [
  {
    id: 'family',
    defaultWidth: '150px',
    label: label[0],
    align: 'left',
    type: 'single',
    columnData: null,
    noData: false,
    disablePadding: true,
    show: true,
    sortable: false,
    defaultFilter: false,
    filter: false,
    custom: false,
  },
  {
    id: 'id',
    defaultWidth: '150px',
    label: label[1],
    align: 'left',
    type: 'single',
    columnData: null,
    noData: false,
    disablePadding: false,
    show: true,
    sortable: true,
    filter: false,
    defaultFilter: false,
    custom: false,
  },
  {
    id: 'survey',
    defaultWidth: '123px',
    type: 'multi',
    label: label[2],
    filterWidth: 170,
    align: 'left',
    columnData: surveyData,
    disablePadding: true,
    noData: false,
    show: true,
    sortable: true,
    filter: true,
    defaultFilter: true,
    custom: false,
  },
  {
    id: 'surveyor',
    defaultWidth: '123px',
    type: 'multi',
    label: label[3],
    filterWidth: 150,
    align: 'left',
    columnData: surveyorData,
    noData: false,
    disablePadding: true,
    show: true,
    sortable: true,
    filter: true,
    defaultFilter: true,
    custom: false,
  },
  {
    id: 'community',
    type: 'levels',
    label: label[4],
    filterWidth: 150,
    align: 'left',
    noData: false,
    columnData: communitiesData,
    disablePadding: true,
    show: true,
    sortable: false,
    filter: true,
    defaultFilter: true,
    custom: false,
  },
  {
    id: 'completed',
    defaultWidth: '100px',
    type: 'date',
    label: label[5],
    columnData: completedData,
    filterWidth: 150,
    align: 'left',
    noData: true,
    disablePadding: true,
    show: true,
    sortable: true,
    filter: true,
    defaultFilter: false,
    custom: false,
  },
];
