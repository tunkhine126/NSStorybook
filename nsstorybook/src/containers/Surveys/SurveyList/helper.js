import React from 'react';
import clsx from 'clsx';
import moment from 'moment';

import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';

import LinkRouter from 'components/shared/LinkRouter';
import { ROUTES } from 'utils/globalConstants';

export function formatData(
  name,
  id,
  questionsCount,
  availableContexts,
  submissionsCount,
  lastUpdate
) {
  return {
    name,
    id,
    questionsCount,
    availableContexts,
    submissionsCount,
    lastUpdate,
  };
}

export const formattedColumns = classes => [
  {
    direction: 'column',
    justify: 'flex-start',
    alignItems: 'flex-start',
    columnId: 'name',
    loadingRows: [{}, {}],
    content(row) {
      return (
        <>
          <Grid item>
            <LinkRouter
              color="primary"
              className={clsx(classes.surveyTemplate, 'recipientColumn')}
              to={ROUTES.surveys.children[1].path.replace(':id', row.id)}
            >
              {row.name}
            </LinkRouter>
          </Grid>
          <Grid item className={classes.familyCommunity}>
            <Typography variant="caption" className={classes.caption}>
              {row.id}
            </Typography>
          </Grid>
        </>
      );
    },
  },
  {
    direction: 'row',
    justify: 'center',
    alignItems: 'flex-start',
    columnId: 'questionsCount',
    loadingRows: [{}],
    content(row) {
      return (
        <Grid item className={classes.columnHeight}>
          {row.questionsCount}
        </Grid>
      );
    },
  },
  {
    direction: 'row',
    justify: 'flex-start',
    alignItems: 'flex-start',
    columnId: 'availableContexts',
    loadingRows: [{}],
    content(row) {
      return (
        <Grid item className={classes.columnHeight}>
          {row.availableContexts}
        </Grid>
      );
    },
  },
  {
    direction: 'row',
    justify: 'center',
    alignItems: 'flex-start',
    columnId: 'submissionsCount',
    loadingRows: [{}],
    content(row) {
      return (
        <Grid item className={classes.columnHeight}>
          {row.submissionsCount}
        </Grid>
      );
    },
  },
  {
    direction: 'row',
    justify: 'flex-start',
    alignItems: 'flex-start',
    columnId: 'lastUpdate',
    loadingRows: [{}],
    content(row) {
      return (
        <Grid item className={classes.columnHeight}>
          {moment(row.lastUpdate).format('MMMM DD, YYYY')}
        </Grid>
      );
    },
  },
];

export const columns = (label, { contextsData, lastUpdateData }) => [
  {
    id: 'name',
    defaultWidth: '300px',
    label: label[0],
    align: 'left',
    type: 'single',
    columnData: null,
    noData: false,
    disablePadding: true,
    show: true,
    sortable: true,
    defaultFilter: false,
    filter: false,
    custom: false,
  },
  {
    id: 'questionsCount',
    defaultWidth: '100px',
    type: 'multi',
    label: label[1],
    align: 'center',
    columnData: null,
    noData: false,
    disablePadding: true,
    show: true,
    sortable: false,
    filter: false,
    defaultFilter: false,
    custom: false,
  },
  {
    id: 'availableContexts',
    defaultWidth: '300px',
    label: label[2],
    filterWidth: 170,
    align: 'left',
    type: 'multi',
    columnData: contextsData,
    noData: false,
    disablePadding: false,
    show: true,
    sortable: false,
    filter: true,
    defaultFilter: false,
    custom: false,
  },
  {
    id: 'submissionsCount',
    defaultWidth: '100px',
    type: 'multi',
    label: label[3],
    align: 'center',
    columnData: null,
    noData: false,
    disablePadding: true,
    show: true,
    sortable: false,
    filter: false,
    defaultFilter: false,
    custom: false,
  },
  {
    id: 'lastUpdate',
    defaultWidth: '200px',
    type: 'date',
    label: label[4],
    columnData: lastUpdateData,
    filterWidth: 150,
    align: 'center',
    noData: true,
    disablePadding: true,
    show: true,
    sortable: false,
    filter: true,
    defaultFilter: true,
    custom: false,
  },
];
