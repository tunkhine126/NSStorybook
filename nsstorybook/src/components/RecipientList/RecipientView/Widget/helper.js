/* eslint-disable no-restricted-syntax */
/* eslint-disable react/display-name */
import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

import InputValidation from 'components/shared/InputValidation';
import DatePickerValidation from 'components/shared/DatePickerValidation';
import TimePickerValidation from 'components/shared/TimePickerValidation';
import SelectValidation from 'components/shared/SelectValidation';
import MultiSelectValidation from 'components/shared/MultiSelectValidation';
import Photo from './fieldTypes/photo';

import { SelectWrapper } from '../../styles';
import { FieldsProvider } from '../context/fields';
import SwitchMode from './switchMode';

export const ftSwitch = (
  field,
  { customFields, uuid },
  key,
  rData,
  handleOptionChange,
  handleDateTimeChange,
  updateRecipient
) => {
  let optionValues = [];
  let optionIntialValue = {};
  let optionIsNull = false;
  const enLabel = field.contexts.us_en.label;

  if (field.option_values) {
    optionValues = field.option_values.map(i => ({ value: i }));

    if (!rData.currentValue[field.name].item) {
      optionIsNull = true;
    }

    optionIntialValue = !optionIsNull
      ? rData.currentValue[field.name].item
      : optionValues[0].value;
  }

  const Field = ({
    classes,
    label,
    name,
    content,
    element,
    fieldType,
    type,
  }) => (
    <FieldsProvider type={type}>
      <SwitchMode
        classes={classes}
        rData={rData}
        field={field}
        uuid={uuid}
        fieldType={fieldType}
        handleDateTimeChange={handleDateTimeChange}
        handleOptionChange={handleOptionChange}
        updateRecipient={updateRecipient}
        label={label}
        currentField={name}
        content={content}
        element={element}
      />
    </FieldsProvider>
  );

  Field.propTypes = {
    classes: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    element: PropTypes.any.isRequired,
    content: PropTypes.any,
    fieldType: PropTypes.string,
    type: PropTypes.string,
  };

  switch (field.field_type) {
    case 'short_text':
      return (
        <Field
          key={key}
          classes="field-container"
          label={enLabel}
          name={field.name}
          content={customFields[field.name]}
          element={
            <InputValidation
              maxWidth="none"
              field={field.name}
              type="text"
              initialValue={customFields[field.name]}
              label={enLabel}
            />
          }
        />
      );
    case 'paragraph_text':
      return (
        <Field
          key={key}
          classes="paragraph-text field-container"
          label={enLabel}
          name={field.name}
          content={customFields[field.name]}
          element={
            <InputValidation
              field={field.name}
              maxWidth="none"
              type="text"
              multiline
              rows="3"
              initialValue={customFields[field.name]}
              label={enLabel}
            />
          }
        />
      );
    case 'number':
      return (
        <Field
          key={key}
          classes="field-container"
          label={enLabel}
          name={field.name}
          content={
            customFields[field.name] ? customFields[field.name].toString() : ''
          }
          element={
            <InputValidation
              type="number"
              maxWidth="none"
              field={field.name}
              initialValue={
                customFields[field.name]
                  ? customFields[field.name].toString()
                  : ''
              }
              label={enLabel}
            />
          }
        />
      );
    case 'date':
      return (
        <Field
          key={key}
          classes="field-container"
          label={enLabel}
          name={field.name}
          content={rData.dates[field.name]}
          element={
            <DatePickerValidation
              field={field.name}
              label={enLabel}
              initialValue={rData.dates[field.name]}
              customFormat="MM-dd-yyyy"
              placeholder={`${moment().format('MM-DD-YYYY')}`}
              variant="dialog"
              inputVariant="filled"
              margin="normal"
              fullWidth
              clearable
              disableFuture
            />
          }
        />
      );
    case 'time':
      return (
        <Field
          key={key}
          classes="field-container"
          label={enLabel}
          name={field.name}
          content={
            rData.times[field.name]
              ? moment(new Date(rData.times[field.name])).format('hh:mm a')
              : null
          }
          fieldType="time"
          element={
            <TimePickerValidation
              field={field.name}
              label={enLabel}
              initialValue={rData.times[field.name]}
              placeholder="08:00 AM"
              inputVariant="filled"
              margin="normal"
              fullWidth
              clearable
            />
          }
        />
      );
    case 'single_select':
      return (
        <Field
          key={key}
          classes="field-container"
          label={enLabel}
          name={field.name}
          content={customFields[field.name]}
          element={(localChildData, localOptionChange) => (
            <SelectWrapper>
              <SelectValidation
                field={field.name}
                type="select"
                maxWidth="none"
                label={enLabel}
                values={
                  !optionIsNull
                    ? localChildData.currentValue[field.name]
                    : optionIntialValue
                }
                initialValue={optionIntialValue}
                items={optionValues}
                onChange={v => localOptionChange(v, field.name)}
                padding="25px 18px 10px 13px"
              />
            </SelectWrapper>
          )}
        />
      );
    case 'multi_select':
      return (
        <Field
          key={key}
          classes="field-container"
          label={enLabel}
          name={field.name}
          content={
            Array.isArray(customFields[field.name])
              ? customFields[field.name].join(', ')
              : customFields[field.name]
          }
          element={(localChildData, localOptionChange) => (
            <SelectWrapper>
              <MultiSelectValidation
                field={field.name}
                type="select"
                maxWidth="none"
                label={enLabel}
                values={
                  !optionIsNull
                    ? localChildData.currentValue[field.name]
                    : optionIntialValue
                }
                initialValue={optionIntialValue}
                onChange={v => localOptionChange(v, field.name)}
                items={optionValues}
                padding="25px 18px 10px 13px"
              />
            </SelectWrapper>
          )}
        />
      );
    case 'likert':
      return (
        <Field
          key={key}
          classes="field-container"
          label={enLabel}
          name={field.name}
          content={customFields[field.name]}
          element={(localChildData, localOptionChange) => (
            <SelectWrapper>
              <SelectValidation
                field={field.name}
                type="select"
                maxWidth="none"
                label={enLabel}
                values={
                  !optionIsNull
                    ? localChildData.currentValue[field.name]
                    : optionIntialValue
                }
                initialValue={optionIntialValue}
                items={optionValues}
                onChange={v => localOptionChange(v, field.name)}
                padding="25px 18px 10px 13px"
              />
            </SelectWrapper>
          )}
        />
      );
    case 'photo':
      return (
        <Field
          type="photo"
          key={key}
          classes="field-container"
          label={enLabel}
          name={field.name}
          content={
            <Photo
              field={field}
              label={enLabel}
              values={customFields}
              recipientData={rData}
            />
          }
          element={
            <Photo
              className="photo-edit-mode"
              rData={rData}
              editMode
              field={field}
              label={enLabel}
              values={customFields}
              recipientData={rData}
            />
          }
        />
      );
    case 'audio':
      return 'audio';
    case 'video':
      return 'video';
    case 'coordinates':
      return (
        <Field
          key={key}
          classes="field-container"
          label={enLabel}
          name={field.name}
          content={customFields[field.name]}
          element={
            <InputValidation
              maxWidth="none"
              field={field.name}
              type="text"
              initialValue={customFields[field.name]}
              label={enLabel}
            />
          }
        />
      );
    default:
      return null;
  }
};
