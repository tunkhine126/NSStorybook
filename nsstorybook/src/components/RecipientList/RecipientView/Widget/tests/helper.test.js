/**
 *
 * Tests for Helper
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import { ftSwitch } from '../helper';
import { RecipientViewProvider } from '../../context';
import { FieldsProvider } from '../../context/fields';

describe('ftSwitch()', () => {
  const field = {
    contexts: { us_en: { label: 'Wake Up Time' } },
    field_type: 'time',
    name: 'wake_up_time',
    option_values: null,
  };
  const paragraphField = {
    contexts: { us_en: { label: 'Name' } },
    field_type: 'paragraph_text',
    name: 'paragraph_text',
    option_values: null,
  };
  const shortTextField = {
    contexts: { us_en: { label: 'Last' } },
    field_type: 'short_text',
    name: 'short_text',
    option_values: null,
  };
  const numberField = {
    contexts: { us_en: { label: 'Age' } },
    field_type: 'number',
    name: 'number',
    option_values: null,
  };

  const rData = {
    currentValue: {
      wake_up_time: {
        item: '',
      },
    },
    dates: {},
    times: {},
  };

  const data = {
    name: 'Test',
    avatarUrl: null,
    internalId: '3erf23-r323d-gewf3-233',
    children: [],
    customFields: {
      uuid: '1c91d190-f5c8-11e9-a1ad-22000be1c01d',
      name: 'Jack Fresh',
      wake_up_time: null,
    },
  };

  const props = {
    data,
    profileDetails: {
      recipientName: 'Country',
      childRecipients: 'Communities',
    },
    match: {
      isExact: true,
      params: { id: '07c538ec-f5c8-11e9-a1ad-22000be1c01d' },
      path: '/families/family-list/family/:id',
      url: '/families/family-list/family/07c538ec-f5c8-11e9-a1ad-22000be1c01d',
    },
    rdData: {},
    tabChange: () => {},
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <RecipientViewProvider {...props} primaryRecipient>
          <FieldsProvider>
            {ftSwitch(field, data, 1, rData, () => {}, () => {}, () => {})}
          </FieldsProvider>
        </RecipientViewProvider>
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should Render Short Text Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <RecipientViewProvider {...props} primaryRecipient>
          <FieldsProvider>
            {ftSwitch(
              shortTextField,
              data,
              1,
              rData,
              () => {},
              () => {},
              () => {}
            )}
          </FieldsProvider>
        </RecipientViewProvider>
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should Render Number Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <RecipientViewProvider {...props} primaryRecipient>
          <FieldsProvider>
            {ftSwitch(
              numberField,
              data,
              1,
              rData,
              () => {},
              () => {},
              () => {}
            )}
          </FieldsProvider>
        </RecipientViewProvider>
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <RecipientViewProvider {...props} primaryRecipient={false}>
          <FieldsProvider>
            {ftSwitch(
              paragraphField,
              data,
              1,
              rData,
              () => {},
              () => {},
              () => {}
            )}
          </FieldsProvider>
        </RecipientViewProvider>
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
