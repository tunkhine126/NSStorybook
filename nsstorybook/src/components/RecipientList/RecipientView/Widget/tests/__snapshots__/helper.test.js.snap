// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ftSwitch() Should render and match the snapshot 1`] = `
<span
  class="paragraph-text field-container"
>
  <div
    class="MuiGrid-root recipient-column MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
    >
      <div
        class="MuiGrid-root makeStyles-captionSpacing-753 MuiGrid-item"
      >
        <span
          class="MuiTypography-root makeStyles-caption-750 MuiTypography-caption"
        >
          Name
        </span>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <p
          class="MuiTypography-root makeStyles-contentBody-752 MuiTypography-body2"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root column-action-btns makeStyles-columnActionBtns-747 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
      >
        <span
          class="MuiGrid-root makeStyles-editBtn-748 MuiGrid-item"
        >
          <span
            class="edit-recipient"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-930 makeStyles-secondary-932"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </span>
        </span>
        <span
          class="MuiGrid-root MuiGrid-item"
        >
          <span
            class="access-time"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-930 makeStyles-primary-931"
              id="paragraph_text_undefined"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                  <path
                    d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </span>
        </span>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      />
    </div>
  </div>
</span>
`;
